src/bsoncxx/test/json.cpp

```diff
@@ -70,4 +70,40 @@ TEST_CASE("CXX-941 is resolved") {
     REQUIRE(output ==
             "{ \"obj_name\" : \"{\\\"id1\\\":\\\"val1\\\", \\\"id2\\\":\\\"val2\\\"}\" }");
 }
+
+TEST_CASE("CXX-1246: Legacy Extended JSON (Implicit)") {
+    bsoncxx::types::b_decimal128 decimal_val("-1234E+999");
+    auto doc = make_document(kvp("number", 42), kvp("dec128", decimal_val));
+    auto output = bsoncxx::to_json(doc.view());
+    REQUIRE(output ==
+            R"({ "number" : 42, "dec128" : { "$numberDecimal" : "-1.234E+1002" } })");
+}
+
+TEST_CASE("CXX-1246: Legacy Extended JSON (Explicit)") {
+    using namespace bsoncxx;
+    types::b_decimal128 decimal_val("-1234E+999");
+    auto doc = make_document(kvp("number", 42), kvp("dec128", decimal_val));
+    auto output = to_json(doc.view(), ExtendedJsonMode::k_legacy);
+    REQUIRE(output ==
+            R"({ "number" : 42, "dec128" : { "$numberDecimal" : "-1.234E+1002" } })");
+}
+
+TEST_CASE("CXX-1246: Relaxed Extended JSON") {
+    using namespace bsoncxx;
+    types::b_decimal128 decimal_val("-1234E+999");
+    auto doc = make_document(kvp("number", 42), kvp("dec128", decimal_val));
+    auto output = to_json(doc.view(), ExtendedJsonMode::k_relaxed);
+    REQUIRE(output ==
+            R"({ "number" : 42, "dec128" : { "$numberDecimal" : "-1.234E+1002" } })");
```

Comment:
Maybe I'm just having trouble seeing it, but what is the difference of the output here on line 97 from that on line 88?

If they don't actually differ, then I'd suggest finding a test case for which they do, assuming there is one.

Reply:
@derickr perhaps a better example of Legacy vs Relaxed might be `$regex`, as that is called out explicitly in the  [Extended JSON spec](https://github.com/mongodb/specifications/blob/master/source/extended-json.rst)